web/src/components/ProjectScreen.js

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import styled from "@emotion/styled";
+import useTasks from "../graphql/useTasks";
+import TaskContent from "./TaskContent";
+import TaskDetailModal from "./TaskDetailModal";
+import EditPermissionsModal from "./EditPermissionsModal";
+import Card from "./Card";
+import Button from "@leafygreen-ui/button";
+import TextInput from "@leafygreen-ui/text-input";
+import { uiColors } from "@leafygreen-ui/palette";
+
+export default function ProjectScreen({
+  currentProject,
+  isEditingPermissions,
+  setIsEditingPermissions,
+}) {
+  return (
+    <Container>
+      {currentProject && <TaskList currentProject={currentProject} />}
+      <EditPermissionsModal
+        project={currentProject}
+        isEditingPermissions={isEditingPermissions}
+        setIsEditingPermissions={setIsEditingPermissions}
+      />
+    </Container>
+  );
+}
+
+const Container = styled.div`
+  height: 100vh;
+  justify-content: center;
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  grid-area: main;
+  background: ${uiColors.gray.light2};
+`;
+
+function useDraftTask({ addTask }) {
+  const [draftTask, setDraftTask] = React.useState(null);
+  const createDraftTask = () => {
+    setDraftTask({ name: "" });
+  };
+  const setDraftTaskName = (name) => {
+    setDraftTask({ name });
+  };
+  const submitDraftTask = async () => {
+    await addTask(draftTask);
+    setDraftTask(null);
+  };
+  return { draftTask, createDraftTask, setDraftTaskName, submitDraftTask };
+}
+
+function TaskList({ currentProject }) {
+  const { tasks, addTask } = useTasks(currentProject);
+  const getTaskById = (id) => tasks.find((task) => task._id === id);
+  const [selectedTaskId, setSelectedTaskId] = React.useState(null);
+  const selectedTask = getTaskById(selectedTaskId);
+
+  const { draftTask, createDraftTask, setDraftTaskName, submitDraftTask } = useDraftTask({ addTask })
+
+  return (
+    <>
+      <List>
+        {tasks.map((task) => (
+          <ListItem key={task._id}>
+            <Card onClick={() => setSelectedTaskId(task._id)}>
+              <TaskContent task={task} />
+            </Card>
+          </ListItem>
+        ))}
+        {draftTask && (
+          <ListItem>
+            <Card>
+              <TextInput
+                type="text"
+                placeholder="Do the dishes"
+                onChange={(e) => {
+                  setDraftTaskName(e.target.value);
+                }}
+                value={draftTask.name}
+              />
+              <Button
+                onClick={() => {
+                  submitDraftTask()
+                }}
+              >
+                Add
+              </Button>
+            </Card>
+          </ListItem>
+        )}
+        <ListItem>
+          <Card>
+            <Button onClick={() => createDraftTask()}>Add Task</Button>
```

Comment:
This "Add Task" button that makes the add task input/button visible is a bit redundant, considering when you press this you then have a "Add Task" and "Add" button. Can you make it so that both buttons are not visible at the same time? Perhaps you can remove the "Add" button altogether? Or alternatively, make it so that the input/add button is an overlay so it doesn't look as redundant in the UI?


<img width="473" alt="Screen Shot 2020-09-17 at 1 30 22 PM" src="https://user-images.githubusercontent.com/52428905/93506930-f71b0400-f8ea-11ea-9e0e-c7608176677f.png">

Reply:
Indeed! Updated to only show the draft card _or_ the add task button, not both at the same time. Also added a cancel button to the draft :)